packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -644,6 +646,10 @@ InternalModel.prototype = {
       }
     }
 
+    if (isEmpty(Object.keys(errors))) {
+      this.send('becameInvalid');
+    }
```

Comment:
What happens if this code runs to put the record in the invalid state then an error is added to the errors object?